code/modules/antagonists/traitor/objectives/demoralise_graffiti.dm

```diff
@@ -0,0 +1,335 @@
+	var/drawing_rune = FALSE
+	/// Set to true if we finished drawing something, this spraycan is now useless
+	var/expended = FALSE
+
+/obj/item/traitor_spraycan/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+
+	if (expended)
+		user.balloon_alert(user, "all out of paint...")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (drawing_rune)
+		user.balloon_alert(user, "already busy!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (!proximity || !check_allowed_items(target) || !isliving(user))
+		return
+
+	if (isturf(target))
+		try_draw_new_rune(user, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	else if (target == rune)
+		try_complete_rune(user, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/**
+ * Attempt to draw a rune on [target_turf].
+ * Shamelessly adapted from the heretic rune drawing process.
+ *
+ * Arguments
+ * * user - the mob drawing the rune
+ * * target_turf - the place the rune's being drawn
+ */
+/obj/item/traitor_spraycan/proc/try_draw_new_rune(mob/living/user, turf/target_turf)
+	if (rune)
+		user.balloon_alert(user, "you've already started drawing somewhere else!")
+		return
+
+	var/list/target_turfs = RANGE_TURFS(1, target_turf)
+	for(var/turf/nearby_turf as anything in target_turfs)
+		if (!isopenturf(nearby_turf) || is_type_in_typecache(nearby_turf, no_draw_turfs))
+			user.balloon_alert(user, "you need a clear 3x3 area!")
+			return
+
+	draw_rune(user, target_turf, target_turfs)
+
+#define RUNE_DRAW_STEP 3 SECONDS
+
+/**
+ * Draw your stage one rune on the ground and store it.
+ *
+ * Arguments
+ * * user - the mob drawing the rune
+ * * target_turf - the place the rune's being drawn
+ */
+/obj/item/traitor_spraycan/proc/draw_rune(mob/living/user, turf/target_turf)
+	user.balloon_alert(user, "drawing outline...")
+	drawing_rune = TRUE
+	if ((try_draw_step("drawing outline...", user, target_turf)))
+		rune = new /obj/effect/decal/cleanable/traitor_rune(target_turf)
+		rune.owner = user
+		RegisterSignal(rune, COMSIG_PARENT_QDELETING, .proc/on_rune_destroyed)
+		try_complete_rune(user, rune)
+
+/**
+ * Called when the rune we were drawing is cleaned up.
+ * If you haven't reached the final stage yet this is fine and you can start over.
+ */
+/obj/item/traitor_spraycan/proc/on_rune_destroyed()
+	SIGNAL_HANDLER
+	UnregisterSignal(rune, COMSIG_PARENT_QDELETING)
+	rune = null
+
+/**
+ * Holder for repeated code to do something after a message and a set amount of time.
+ *
+ * Arguments
+ * * output - a string to show when you start the process
+ * * user - the mob drawing the rune
+ * * target - what they're trying to draw, or the place they are trying to draw on
+ */
+/obj/item/traitor_spraycan/proc/try_draw_step(start_output, mob/living/user, atom/target)
+	user.balloon_alert(user, "[start_output]")
+	if (!do_after(user, RUNE_DRAW_STEP, target))
+		user.balloon_alert(user, "interrupted!")
+		drawing_rune = FALSE
+		return FALSE
+
+	spray_sound()
+	drawing_rune = FALSE
+	return TRUE
```

Comment:
should you really set drawing_rune to false here? isn't it only set to true at the start?